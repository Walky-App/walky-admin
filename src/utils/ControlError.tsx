import type { FieldErrors } from 'react-hook-form'

export function getFormErrorMessage(path: string, errors: FieldErrors) {
  const pathParts = path.split('.')
  let error: FieldErrors = errors

  for (const part of pathParts) {
    if (typeof error !== 'object' || error === null) {
      return null
    }
    error = error[part as keyof typeof error] as FieldErrors
  }

  if (error?.message) {
    return error.message ? <p className="mt-2 text-sm text-red-600">{String(error.message)}</p> : null
  }

  return null
}
