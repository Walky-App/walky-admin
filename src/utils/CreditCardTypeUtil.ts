export const getCardType = (number: string) => {
  const firstDigit = number[0]
  const firstTwoDigits = number.slice(0, 2)
  const firstFourDigits = number.slice(0, 4)

  if (/^4/.test(firstDigit)) {
    return 'Visa'
  } else if (/^5[1-5]/.test(firstTwoDigits)) {
    return 'MasterCard'
  } else if (/^3[47]/.test(firstTwoDigits)) {
    return 'Amex'
  } else if (/^6(?:011|5)/.test(firstFourDigits)) {
    return 'Discover'
  } else {
    return 'Unknown'
  }
}
