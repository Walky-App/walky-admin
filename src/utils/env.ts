export const getEnv = () => {
  return {
    VITE_API_BASE_URL: import.meta.env.VITE_API_BASE_URL || 'https://staging.walkyapp.com/api'
  };
};
