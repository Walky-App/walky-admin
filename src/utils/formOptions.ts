export type INotificationPreference = 'notification_email' | 'notification_sms'

export const notifications_preferences: INotificationPreference[] = ['notification_email', 'notification_sms']

export const job_preferences = [
  'Harvester',
  'Budtender',
  'Trimmer',
  'Packager',
  'Gardener',
  'Cultivator',
  'Extractor',
  'Front desk',
  'Greeter',
  'Id checker',
  'Inventory',
  'Data entry',
  'Event staff',
  'Promo representative',
  'Cleaning',
  'Joint roller',
  'Grow tech',
  'Clone tech',
  'Sign spinner',
]

export const facilityContactRoles = [
  { label: 'Owner', value: 'Owner' },
  { label: 'AP', value: 'AP' },
  { label: 'Onsite', value: 'Onsite' },
  { label: 'Security', value: 'Security' },
  { label: 'Manager', value: 'Manager' },
  { label: 'Other', value: 'Other' },
]

export const services = ['Trimming', 'Harvesting', 'Packaging', 'Budtending', 'Gardening', 'General Labor', 'Other']

export const countries = [
  { label: 'United States', value: 'US' },
  { label: 'Canada', value: 'CA' },
  { label: 'Mexico', value: 'MX' },
]

export const states = [
  { label: 'Alabama', value: 'AL' },
  { label: 'Alaska', value: 'AK' },
  { label: 'Arizona', value: 'AZ' },
  { label: 'Arkansas', value: 'AR' },
  { label: 'California', value: 'CA' },
  { label: 'Colorado', value: 'CO' },
  { label: 'Connecticut', value: 'CT' },
  { label: 'Delaware', value: 'DE' },
  { label: 'Florida', value: 'FL' },
  { label: 'Georgia', value: 'GA' },
  { label: 'Hawaii', value: 'HI' },
  { label: 'Idaho', value: 'ID' },
  { label: 'Illinois', value: 'IL' },
  { label: 'Indiana', value: 'IN' },
  { label: 'Iowa', value: 'IA' },
  { label: 'Kansas', value: 'KS' },
  { label: 'Kentucky', value: 'KY' },
  { label: 'Louisiana', value: 'LA' },
  { label: 'Maine', value: 'ME' },
  { label: 'Maryland', value: 'MD' },
  { label: 'Massachusetts', value: 'MA' },
  { label: 'Michigan', value: 'MI' },
  { label: 'Minnesota', value: 'MN' },
  { label: 'Mississippi', value: 'MS' },
  { label: 'Missouri', value: 'MO' },
  { label: 'Montana', value: 'MT' },
  { label: 'Nebraska', value: 'NE' },
  { label: 'Nevada', value: 'NV' },
  { label: 'New Hampshire', value: 'NH' },
  { label: 'New Jersey', value: 'NJ' },
  { label: 'New Mexico', value: 'NM' },
  { label: 'New York', value: 'NY' },
  { label: 'North Carolina', value: 'NC' },
  { label: 'North Dakota', value: 'ND' },
  { label: 'Ohio', value: 'OH' },
  { label: 'Oklahoma', value: 'OK' },
  { label: 'Oregon', value: 'OR' },
  { label: 'Pennsylvania', value: 'PA' },
  { label: 'Rhode Island', value: 'RI' },
  { label: 'South Carolina', value: 'SC' },
  { label: 'South Dakota', value: 'SD' },
  { label: 'Tennessee', value: 'TN' },
  { label: 'Texas', value: 'TX' },
  { label: 'Utah', value: 'UT' },
  { label: 'Vermont', value: 'VT' },
  { label: 'Virginia', value: 'VA' },
  { label: 'Washington', value: 'WA' },
  { label: 'West Virginia', value: 'WV' },
  { label: 'Wisconsin', value: 'WI' },
  { label: 'Wyoming', value: 'WY' },
]

export const jobTitlesOptions = [
  { title: 'Harvester', value: 'Harvester' },
  { title: 'Budtender', value: 'Budtender' },
  { title: 'Trimmer', value: 'Trimmer' },
  { title: 'Packager', value: 'Packager' },
  { title: 'Extractor', value: 'Extractor' },
  { title: 'Front desk', value: 'Front desk' },
  { title: 'Greeter', value: 'Greeter' },
  { title: 'Id checker', value: 'Id checker' },
  { title: 'Inventory', value: 'Inventory' },
  { title: 'Data entry', value: 'Data entry' },
  { title: 'Event staff', value: 'Event staff' },
  { title: 'Promo representative', value: 'Promo representative' },
  { title: 'Cleaning', value: 'Cleaning' },
  { title: 'Joint roller', value: 'Joint roller' },
  { title: 'Grow tech', value: 'Grow tech' },
  { title: 'Clone tech', value: 'Clone tech' },
  { title: 'Sign spinner', value: 'Sign spinner' },
]

export const lunchTimeOptions = [
  { label: '30 minutes', value: 30 },
  { label: '45 minutes', value: 45 },
  { label: '60 minutes', value: 60 },
]

export const jobTipsOptions = [
  { label: 'Change Required Upon Entry', value: 'Change Required Upon Entry' },
  { label: 'Lunch Included', value: 'Lunch Included' },
  { label: 'Lunch Room Available', value: 'Lunch Room Available' },
  { label: 'Lunch Will Be Off-Premise', value: 'Lunch Will Be Off-Premise' },
  { label: 'Pack a Lunch', value: 'Pack a Lunch' },
  { label: 'Parking on Street', value: 'Parking on Street' },
  { label: 'Parking Onsite', value: 'Parking Onsite' },
  { label: 'Required Identification', value: 'Required Identification' },
  { label: 'Special Equipment', value: 'Special Equipment' },
  { label: 'No gas stations nearby', value: 'No gas stations nearby' },
  { label: 'Water is provided', value: 'Water is provided' },
  { label: 'Outdoor sun exposure', value: 'Outdoor sun exposure' },
  { label: 'Must be able to lift 50 lbs', value: 'Must be able to lift 50 lbs' },
  { label: 'Steel toe shoes', value: 'Steel toe shoes' },
  { label: 'Lab coat Provided', value: 'Lab coat Provided' },
  { label: 'Head / Beard net required', value: 'Head / Beard net required' },
]
