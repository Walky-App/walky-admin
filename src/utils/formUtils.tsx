import { type FieldErrors } from 'react-hook-form'

export function getFormErrorMessage(path: string, errors: FieldErrors) {
  const pathParts = path.split('.')
  let error: FieldErrors = errors

  for (const part of pathParts) {
    if (typeof error !== 'object' || error === null) {
      return null
    }
    error = error[part as keyof typeof error] as FieldErrors
  }

  if (error?.message) {
    return <p className="mt-2 text-sm text-red-600">{String(error.message)}</p>
  }

  return null
}

export const requiredFieldsNoticeText = (
  <p className="mt-1 text-sm leading-6 text-gray-600">
    <span style={{ color: 'red' }}>*</span> indicates a required field.
  </p>
)
