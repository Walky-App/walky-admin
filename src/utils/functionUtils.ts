/* eslint-disable @typescript-eslint/no-explicit-any */
export function secondsToTimeDescription(seconds: number) {
  const minutes = Math.floor(seconds / 60)
  const hours = Math.floor(minutes / 60)

  if (hours >= 1 && minutes % 60 === 0) {
    return `${hours} hour${hours > 1 ? 's' : ''}`
  } else if (hours >= 1 && minutes % 60 !== 0) {
    const remainingMinutes = minutes % 60
    return `${hours}.${remainingMinutes} hours`
  } else {
    return `${minutes} minute${minutes !== 1 ? 's' : ''}`
  }
}

export function getModifiedProperties(oldObj: any, newObj: any): Partial<any> {
  const modifiedProperties: Partial<any> = {}
  for (const key of Object.keys(newObj)) {
    if (newObj[key] !== oldObj[key]) {
      modifiedProperties[key] = newObj[key]
    }
  }
  return modifiedProperties
}
