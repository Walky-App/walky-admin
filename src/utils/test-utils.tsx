import React, { type ReactElement } from 'react'

import { MemoryRouter } from 'react-router-dom'

import '@testing-library/jest-dom'
import { render, type RenderOptions } from '@testing-library/react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  const originalConsoleError = console.error
  const jsDomCssError = 'Error: Could not parse CSS stylesheet'
  console.error = (...params) => {
    if (!params.find(p => p.toString().includes(jsDomCssError))) {
      originalConsoleError(...params)
    }
  }
  return <MemoryRouter>{children}</MemoryRouter>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
