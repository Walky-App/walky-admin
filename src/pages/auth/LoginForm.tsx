import { useState } from 'react'

import { useLocation, useNavigate } from 'react-router-dom'

import { Button } from 'primereact/button'
import { InputText } from 'primereact/inputtext'
import { Password } from 'primereact/password'

import { useAuth } from '../../contexts/AuthContext'
import { type LoginData } from '../../interfaces/global'
import { type ILoginData } from '../../interfaces/loginData'
import { type ITokenInfo } from '../../interfaces/services'
import { LoginService } from '../../services/authService'
import { useUtils } from '../../store/useUtils'
import { roleChecker } from '../../utils/roleChecker'
import { GetTokenInfo, SetToken } from '../../utils/tokenUtil'

export const LoginForm = ({ setUserForm }: { setUserForm: (value: string) => void }) => {
  const [error, setError] = useState<Error>()
  const [loading, setLoading] = useState(false)
  const [value, setValue] = useState<string>('')
  const { setAvatarImageUrl } = useUtils()

  const location = useLocation()

  const params = new URLSearchParams(location.search)
  const redirectPath = params.get('redirect')

  const { setUser } = useAuth()
  const navigate = useNavigate()
  const tokenInfo = GetTokenInfo()

  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()
    await setLoading(true)
    const form = event.target as HTMLFormElement

    const body: LoginData = {
      email: form.email.value,
      password: form.password.value,
    }

    const data: ILoginData = await LoginService(body)

    try {
      if (!data) {
        setLoading(false)
        return setError(new Error('Email/Password invalid'))
      } else {
        const { access_token, user }: ILoginData = data
        if (user && access_token) {
          const data: ITokenInfo = {
            first_name: user.first_name,
            _id: user._id,
            role: user.role,
            access_token: access_token,
            state: user.state,
            avatar: user.avatar,
            onboarding: user.onboarding,
          }

          SetToken(data)
          setUser({ ...user, access_token: access_token, onboarding: user.onboarding })
          setAvatarImageUrl(user.avatar as string)
          if (redirectPath) {
            navigate(redirectPath, { replace: true })
          } else {
            switch (roleChecker()) {
              case 'admin':
                navigate('/admin/dashboard')
                break
              case 'client':
                if (tokenInfo.onboarding?.completed === false || tokenInfo.onboarding === undefined) {
                  navigate('/client/onboarding')
                } else {
                  navigate('/client/dashboard')
                }
                break
              case 'employee':
                if (tokenInfo.onboarding?.completed === false || tokenInfo.onboarding === undefined) {
                  navigate('/employee/onboarding')
                } else {
                  navigate('/employee/dashboard')
                }
                break
              case 'sales':
                navigate('/sales/dashboard')
                break
              default:
                navigate('/login')
            }
          }
        } else {
          setError(data.message)
          setLoading(false)
        }
      }
    } catch (error) {
      console.error(error)
    }
  }

  return (
    <form
      onSubmit={handleSubmit}
      className="mb-6 ml-auto mr-auto max-w-md space-y-4 px-4 sm:px-0"
      data-testid="login-form">
      <div className="mx-auto max-w-lg text-center" />
      <div>
        <label htmlFor="email" className="sr-only">
          Email
        </label>

        <div className="relative">
          <InputText
            required
            type="email"
            name="email"
            className="w-full rounded-lg border-zinc-200 p-4 shadow-sm  focus:border-green-500 focus:ring-green-500"
            placeholder="*Email"
          />
          <span className="absolute inset-y-0 end-0 grid place-content-center px-4">
            <svg
              xmlns="http://www.w3.org/2000/svg"
              className="h-4 w-4 text-zinc-700"
              fill="none"
              viewBox="0 0 24 24"
              stroke="currentColor">
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                d="M16 12a4 4 0 10-8 0 4 4 0 008 0zm0 0v1.5a2.5 2.5 0 005 0V12a9 9 0 10-9 9m4.5-1.206a8.959 8.959 0 01-4.5 1.207"
              />
            </svg>
          </span>
        </div>
      </div>

      <Password
        inputId="password"
        placeholder="*Password"
        value={value}
        onChange={e => setValue(e.target.value)}
        toggleMask
        pt={{
          panel: { className: 'hidden' },
          input: {
            className: 'w-full rounded-lg border-zinc-200 p-4 shadow-sm focus:border-green-500 focus:ring-green-500',
          },
          iconField: { root: { className: 'w-full' } },
        }}
        className="w-full"
      />

      {error ? (
        <div className="flex items-center justify-center">
          <p className="text-sm text-red-500">{String(error)}</p>
        </div>
      ) : null}
      <div className="w-full text-right" style={{ marginTop: 0 }}>
        <button
          type="button"
          tabIndex={-1}
          className="text-xs text-zinc-500 underline hover:text-green-700"
          onClick={() => setUserForm('Forgot Password')}>
          Forgot your password?
        </button>
      </div>
      <Button
        label={loading ? 'Logging in' : 'Login'}
        type="submit"
        className={`w-full ${loading && 'cursor-wait hover:bg-zinc-950'}`}
      />
    </form>
  )
}
