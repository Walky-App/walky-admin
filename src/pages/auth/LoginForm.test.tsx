import { Auth } from '.'
import { cleanup, render, screen } from '../../utils/test-utils'
import { LoginForm } from './LoginForm'

describe('Auth Page', () => {
  afterEach(() => {
    cleanup()
  })

  it('should render auth view', () => {
    render(<Auth />)
  })

  it('should match snapshot', () => {
    const { asFragment, container } = render(<Auth />)

    const imageComponent = container.querySelector('[data-testid="hero-image"]')
    imageComponent?.setAttribute('src', 'https://hemptemps-prod.s3.amazonaws.com/web-images/mock-image.png')

    expect(asFragment()).toMatchSnapshot()
  })

  it('should render the login component', async () => {
    render(<LoginForm setUserForm={() => jest.fn()} />).container

    const loginForm = await screen.findByTestId('login-form')
    expect(loginForm).toBeInTheDocument()
  })

  it('should render the email input field', async () => {
    render(<LoginForm setUserForm={() => jest.fn()} />)

    const emailInput = await screen.getByPlaceholderText('*Email')
    expect(emailInput).toBeInTheDocument()
  })
})
