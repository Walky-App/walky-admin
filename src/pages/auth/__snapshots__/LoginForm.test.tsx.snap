// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Page should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="flex h-full min-h-screen items-center justify-center sm:mb-8 md:mb-0 lg:items-center"
    data-testid="auth-section"
  >
    <div
      class="gap-10 md:basis-1/2"
    >
      <div
        class="w-full px-4 sm:px-0"
      >
        <div
          class="flex justify-center"
        >
          <img
            alt="Hemp Temps logo"
            class="sm:w-full xl:px-12"
            src="/assets/logos/logo-horizontal-cropped.png"
          />
        </div>
        <form
          class="mb-6 ml-auto mr-auto max-w-md space-y-4 px-4 sm:px-0"
          data-testid="login-form"
        >
          <div
            class="mx-auto max-w-lg text-center"
          />
          <div>
            <label
              class="sr-only"
              for="email"
            >
              Email
            </label>
            <div
              class="relative"
            >
              <input
                class="w-full rounded-lg border-zinc-200 p-4 shadow-sm  focus:border-green-500 focus:ring-green-500 p-inputtext p-component"
                data-pc-name="inputtext"
                data-pc-section="root"
                name="email"
                placeholder="*Email"
                required=""
                type="email"
              />
              <span
                class="absolute inset-y-0 end-0 grid place-content-center px-4"
              >
                <svg
                  class="h-4 w-4 text-zinc-700"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 12a4 4 0 10-8 0 4 4 0 008 0zm0 0v1.5a2.5 2.5 0 005 0V12a9 9 0 10-9 9m4.5-1.206a8.959 8.959 0 01-4.5 1.207"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            class="w-full p-password p-component p-inputwrapper p-input-icon-right"
            data-pc-name="password"
            data-pc-section="root"
          >
            <div
              class="p-icon-field p-icon-field-right w-full"
              data-pc-name="iconfield"
              data-pc-section="root"
            >
              <span
                class="p-input-icon"
                data-pc-name="inputicon"
                data-pc-section="root"
              >
                <svg
                  aria-checked="true"
                  aria-hidden="true"
                  aria-label="Show Password"
                  class="p-icon p-password-show-icon"
                  data-pc-section="showicon"
                  fill="none"
                  height="14"
                  role="switch"
                  tabindex="0"
                  viewBox="0 0 14 14"
                  width="14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M0.0535499 7.25213C0.208567 7.59162 2.40413 12.4 7 12.4C11.5959 12.4 13.7914 7.59162 13.9465 7.25213C13.9487 7.2471 13.9506 7.24304 13.952 7.24001C13.9837 7.16396 14 7.08239 14 7.00001C14 6.91762 13.9837 6.83605 13.952 6.76001C13.9506 6.75697 13.9487 6.75292 13.9465 6.74788C13.7914 6.4084 11.5959 1.60001 7 1.60001C2.40413 1.60001 0.208567 6.40839 0.0535499 6.74788C0.0512519 6.75292 0.0494023 6.75697 0.048 6.76001C0.0163137 6.83605 0 6.91762 0 7.00001C0 7.08239 0.0163137 7.16396 0.048 7.24001C0.0494023 7.24304 0.0512519 7.2471 0.0535499 7.25213ZM7 11.2C3.664 11.2 1.736 7.92001 1.264 7.00001C1.736 6.08001 3.664 2.80001 7 2.80001C10.336 2.80001 12.264 6.08001 12.736 7.00001C12.264 7.92001 10.336 11.2 7 11.2ZM5.55551 9.16182C5.98308 9.44751 6.48576 9.6 7 9.6C7.68891 9.59789 8.349 9.32328 8.83614 8.83614C9.32328 8.349 9.59789 7.68891 9.59999 7C9.59999 6.48576 9.44751 5.98308 9.16182 5.55551C8.87612 5.12794 8.47006 4.7947 7.99497 4.59791C7.51988 4.40112 6.99711 4.34963 6.49276 4.44995C5.98841 4.55027 5.52513 4.7979 5.16152 5.16152C4.7979 5.52513 4.55027 5.98841 4.44995 6.49276C4.34963 6.99711 4.40112 7.51988 4.59791 7.99497C4.7947 8.47006 5.12794 8.87612 5.55551 9.16182ZM6.2222 5.83594C6.45243 5.6821 6.7231 5.6 7 5.6C7.37065 5.6021 7.72553 5.75027 7.98762 6.01237C8.24972 6.27446 8.39789 6.62934 8.4 7C8.4 7.27689 8.31789 7.54756 8.16405 7.77779C8.01022 8.00802 7.79157 8.18746 7.53575 8.29343C7.27994 8.39939 6.99844 8.42711 6.72687 8.37309C6.4553 8.31908 6.20584 8.18574 6.01005 7.98994C5.81425 7.79415 5.68091 7.54469 5.6269 7.27312C5.57288 7.00155 5.6006 6.72006 5.70656 6.46424C5.81253 6.20842 5.99197 5.98977 6.2222 5.83594Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <input
                class="p-password-input w-full rounded-lg border-zinc-200 p-4 shadow-sm focus:border-green-500 focus:ring-green-500 p-inputtext p-component"
                data-pc-name="inputtext"
                data-pc-section="root"
                id="password"
                placeholder="*Password"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="w-full text-right"
            style="margin-top: 0px;"
          >
            <button
              class="text-xs text-zinc-500 underline hover:text-green-700"
              tabindex="-1"
              type="button"
            >
              Forgot your password?
            </button>
          </div>
          <button
            aria-label="Login"
            class="w-full false p-button p-component"
            data-pc-name="button"
            data-pc-section="root"
            type="submit"
          >
            <span
              class="p-button-label p-c"
              data-pc-section="label"
            >
              Login
            </span>
          </button>
        </form>
        <div
          class="mr-center flex flex-col items-center justify-center gap-2"
        >
          <div>
            <div
              class="relative"
            >
              <div
                aria-hidden="true"
                class="absolute inset-0 flex items-center"
              >
                <div
                  class="w-full border-t border-gray-300"
                />
              </div>
              <div
                class="relative flex justify-center"
              >
                <span
                  class="bg-white px-2 text-sm text-gray-500"
                >
                  OR
                </span>
              </div>
            </div>
            <button
              class="font-medium underline hover:text-green-700 p-button p-component p-button-text"
              data-pc-name="button"
              data-pc-section="root"
            >
              Sign up
            </button>
          </div>
          <div
            class=" text-zinc-500"
          />
        </div>
      </div>
      <footer
        aria-labelledby="footer-heading"
        class="py-6"
      >
        <div
          class="text-center text-xs text-gray-500 md:items-center md:justify-center"
        >
          <p
            class=""
          >
            Copyright Â© 2024 Hemp Temps
          </p>
          <p
            class="my-1"
          >
            <a
              class="underline"
              href="https://hemptemps.com/terms-and-conditions/"
              target="_blank"
            >
              Terms and Conditions
            </a>
          </p>
          <small
            class="text-[10px] font-extralight"
          >
            v1.1.0
          </small>
        </div>
      </footer>
    </div>
    <div
      class="relative hidden w-full sm:h-96 lg:block lg:h-screen lg:basis-1/2"
    >
      <img
        alt="Hemp Temps"
        class="absolute inset-0 hidden h-full object-cover sm:block"
        data-testid="hero-image"
        src="https://hemptemps-prod.s3.amazonaws.com/web-images/mock-image.png"
      />
    </div>
  </section>
</DocumentFragment>
`;
